src/charm.py

```diff
@@ -41,8 +42,9 @@ def __init__(self, *args: typing.Any) -> None:
         except CharmConfigInvalidError as exc:
             self.model.unit.status = ops.BlockedStatus(exc.msg)
             return
-        self._synapse = Synapse(charm_state=self._charm_state)
-        self.pebble_service = PebbleService(synapse=self._synapse)
+        self.synapse = Synapse(charm_state=self._charm_state)
```

Comment:
What is the reason for making this public please?

Reply:
I changed it to public so synapse_api could use it through the charm instance. It is needed to get the field registration_shared_secret from the configuration file.

Reply:
Reverted to private after changing Synapse API to a module.